docs/site/todo-tutorial-controller.md

```diff
@@ -124,25 +124,25 @@ export class TodoController {
     @repository(TodoRepository) protected todoRepo: TodoRepository,
   ) {}
 
-  @post('/todo')
+  @post('/todos')
   async createTodo(@requestBody() todo: Todo) {
     if (!todo.title) {
       throw new HttpErrors.BadRequest('title is required');
     }
     return await this.todoRepo.create(todo);
   }
 
-  @get('/todo/{id}')
+  @get('/todos/{id}')
```

Comment:
Similarly here. In REST, it's common and natural to have two path namespaces for a single model - `/todo` to work with a single model instance, `/todos` to work with the collection.

Such design has an important property: it avoids naming clashes between instance ids and remote method names - for example, it's possible to have `todo` with id `findOne` (available at `GET /todo/findOne`) and an remote method called `findOne` (exposed at `GET /todos/findOne`). This is something we cannot do in LB 3.x.

Reply:
With that in mind, I'm wondering whether we should modify our template to 'correct' the designs of our available controller methods. Putting this into example, if given path url was `todo`, should `findById` be tied to `/todo/{id}` while `find` is tied to `/todos`?

Reply:
> for example, it's possible to have todo with id findOne (available at GET /todo/findOne) and an remote method called findOne (exposed at GET /todos/findOne). This is something we cannot do in LB 3.x.

👍 

> My only concern is about the UX and implementation complexity of such solution

I think now we don't append namespace prefix to the path of controller method, e.g. `@post('/todo')` or `@post('/todos')`, user need to provide the completed path, maybe we can enhance the REST decorators to allow specifying the namespace prefix: 
- plural 
- singula
- customized

Reply:
The UX of this approach does seem difficult to grasp. Here are all of the possible options that may be available to the users:
When given `Todo` model,
- Use `todo` and `todos`
- Use just `todo`
- Use just `todos`
- Custom http path
  - Use only the custom path given
  - Prompt for custom singular path
    - Use plural form of given answer
    - Custom plural form

As you can see, there are tons of options that can be available to the users and if we were to go with this approach, we should only pick a couple of these to avoid overloading users with choices.

@strongloop/lb-next-dev What do you think?

Reply:
my suggestion in https://github.com/strongloop/loopback-next/pull/1314#discussion_r187385245 is totally a different story :) As this PR I would like to just focus on adding the prompt.
For the singular/plural namespace, 
- use singular for the item takes in param with type `Todo`
- use plural for those take in param with type `Todo[]`

Reply:
For reference, this is how we handle this in [LB3](http://loopback.io/doc/en/lb2/Exposing-models-over-REST.html#rest-paths). I also like the idea of following REST principles in our templates. 

>For the singular/plural namespace, my take would be
>- use singular for the item takes in param with type Todo
>- use plural for those take in param with type Todo[]

if this can be easily done, then that would make our UX easier and then we are left with a custom http path prompt. Maybe we can compromise between UX and flexibility.